Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStates.cs

```diff
@@ -27,60 +29,125 @@ public class StreamProcessorStates : IStreamProcessorStateRepository
+        /// <param name="id">The unique <see cref="IStreamProcessorId" /> representing either the <see cref="AbstractScopedStreamProcessor"/>
+        /// or <see cref="SubscriptionId" />.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken" />.</param>
+        /// <returns>A <see cref="Task" /> that, when resolved, returns <see cref="Try{TResult}" />.</returns>
+        public async Task<Try<IStreamProcessorState>> TryGetFor(IStreamProcessorId id, CancellationToken cancellationToken)
         {
             try
             {
-                return Task.FromResult(new Try<IStreamProcessorState>(false, null));
-
-                // TODO: There are now two different collections for stream proecssor states. One for StreamProcessorId (Events.Processing) and one for SubscriptionId (EventHorizon)
-                // var states = await _connection.GetStreamProcessorStateCollection(streamProcessorId.ScopeId, cancellationToken).ConfigureAwait(false);
-                // var persistedState = await states.Find(
-                //     _streamProcessorFilter.Eq(_ => _.ScopeId, streamProcessorId.ScopeId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.EventProcessorId, streamProcessorId.EventProcessorId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.SourceStreamId, streamProcessorId.SourceStreamId.Value))
-                //     .FirstOrDefaultAsync(cancellationToken)
-                //     .ConfigureAwait(false);
-                // if (persistedState != null)
-                // {
-                //     return (true, persistedState.ToRuntimeRepresentation());
-                // }
-                // else
-                // {
-                //     return (false, null);
-                // }
+                if (id is SubscriptionId subscriptionId)
+                {
+                    var states = await _connection.GetSubscriptionStateCollection(subscriptionId.ScopeId, cancellationToken).ConfigureAwait(false);
+                    var persistedState = await states.Find(CreateFilter(subscriptionId))
+                        .FirstOrDefaultAsync(cancellationToken)
+                        .ConfigureAwait(false);
+                    return (persistedState != null) ? (true, persistedState.ToRuntimeRepresentation()) : (false, null);
+                }
+                else
+                {
+                    var streamProcessorId = id as StreamProcessorId;
+                    var states = await _connection.GetStreamProcessorStateCollection(streamProcessorId.ScopeId, cancellationToken).ConfigureAwait(false);
+                    var persistedState = await states.Find(CreateFilter(streamProcessorId))
+                        .FirstOrDefaultAsync(cancellationToken)
+                        .ConfigureAwait(false);
+                    return (persistedState != null) ? (true, persistedState.ToRuntimeRepresentation()) : (false, null);
+                }
```

Comment:
I think we should throw an exception if id is no StreamPRocessorId or SubscriptionId